### DISCUSSION TITLE
lastRead - contradiction?

### DISCUSSION THREAD

**Discussion by barbequesauce** [anansi-project/opds-pse#2] [General]
Hiya! 

Looking at 1.1 of PSE - wondering if this isn’t a contradiction?

The Link MAY contain the following attributes:

lastRead MUST provide the last page read for this document. The numbering starts at **_1_**.
Notes

pages are numbered from **_0_** to N-1 (N being the total number of pages)

**Comment by gotson** [comment]
> pages are numbered from **_0_** to N-1 (N being the total number of pages)

That's for the `href` attribute.



> lastRead MUST provide the last page read for this document. The numbering starts at **_1_**.

The `lastRead` attribute starts at 1, not 0.

**Comment by barbequesauce** [comment]
Yeah I get that - why wouldn't they use the same numbering?

**Reply by gotson** [reply]
the `0 - N-1` is more of an index of a page, and the historical way Tom created this for Ubooquity. It's counter-intuitive though, you don't start a book at page 0, but i guess arrays having a zero-based index is probably the reason why.

for read progress, say that you are at page 5 while it's actually page 6 would be very complex, and prone to errors, so we decided to have a clean slate.

**Comment by barbequesauce** [comment]
Yeah I get the clean slate, but I would also get consistency... IMHO the inconsistency could be confusing for implementors (speaking as an implementor)

**Reply by gotson** [reply]
should we clarify the spec to make this easier to understand, and with more warnings to explain the difference?

**Comment by barbequesauce** [comment]
That would help, I would think.

**Reply by gotson** [reply]
care to send a PR ? I fear my update would be biased since i wrote the thing in the first place 😅

**Comment by barbequesauce** [comment]
Fair enough - let me sleep on it, make sure I'm not too snarky lol
